src/ucp/wireup/wireup_ep.c

```diff
@@ -596,8 +596,9 @@ void ucp_wireup_ep_destroy_next_ep(ucp_wireup_ep_t *wireup_ep)
 
     ucs_assert(wireup_ep != NULL);
     uct_ep = ucp_wireup_ep_extract_next_ep(&wireup_ep->super.super);
-    ucs_assert_always(uct_ep != NULL);
-    uct_ep_destroy(uct_ep);
+    if (uct_ep) {
```

Comment:
1. why changed?
2. uct_ep != NULL

Reply:
I tested and found if got non-zero return value from rdma_resolve_addr(), uct_ep_create() got NULL uct_ep and cored at line ucs_assert_always(uct_ep != NULL)

Reply:
uct_ep_create() will alloc a new uct_ep and init it, if init failed allocated memory is freed and return NULL uct_ep. 
ucp_wireup_ep_set_next_ep() will not be called when uct_ep_create() failed.

Hence added ucp_cm_client_try_fallback_cms() after uct_ep_create() failed will core in ucp_cm_client_try_next_cm_progress() at ucp_wireup_ep_destroy_next_ep().

Reply:
"failed will core" - what is the backtrace?

Reply:
backtrace is this:
[gen-l-vrt-435:190026:0:190026]   wireup_ep.c:599  Assertion `uct_ep != NULL' failed

/.autodirect/swgwork/minf/pvt-ucx1022/src/ucp/wireup/wireup_ep.c: [ ucp_wireup_ep_destroy_next_ep() ]
      ...
      596
      597     ucs_assert(wireup_ep != NULL);
      598     uct_ep = ucp_wireup_ep_extract_next_ep(&wireup_ep->super.super);
==>   599     ucs_assert_always(uct_ep != NULL);
      600     uct_ep_destroy(uct_ep);
      601
      602     wireup_ep->flags &= ~UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;

==== backtrace (tid: 190026) ====
 0 0x00000000000e2efa ucp_wireup_ep_destroy_next_ep()  /.autodirect/swgwork/minf/pvt-ucx1022/src/ucp/wireup/wireup_ep.c:599
 1 0x00000000000ee540 ucp_cm_client_try_next_cm_progress()  /.autodirect/swgwork/minf/pvt-ucx1022/src/ucp/wireup/wireup_cm.c:84
 2 0x00000000000565fb ucs_callbackq_slow_proxy()  /.autodirect/swgwork/minf/pvt-ucx1022/src/ucs/datastruct/callbackq.c:402
 3 0x0000000000054442 ucs_callbackq_dispatch()  /.autodirect/swgwork/minf/pvt-ucx1022/src/ucs/datastruct/callbackq.h:211
 4 0x0000000000054442 uct_worker_progress()  /.autodirect/swgwork/minf/pvt-ucx1022/src/uct/api/uct.h:2600
 5 0x0000000000054442 ucp_worker_progress()  /.autodirect/swgwork/minf/pvt-ucx1022/src/ucp/core/ucp_worker.c:2669
 6 0x0000000000406485 UcxContext::progress_worker_event()  /.autodirect/swgwork/minf/pvt-ucx1022/test/apps/iodemo/ucx_wrapper.cc:388
 7 0x000000000040cdcb UcxContext::progress()  /.autodirect/swgwork/minf/pvt-ucx1022/test/apps/iodemo/ucx_wrapper.cc:226
 8 0x000000000041a18d DemoClient::run()  /.autodirect/swgwork/minf/pvt-ucx1022/test/apps/iodemo/io_demo.cc:2177
 9 0x0000000000404953 do_client()  /.autodirect/swgwork/minf/pvt-ucx1022/test/apps/iodemo/io_demo.cc:2824
10 0x0000000000404953 main()  /.autodirect/swgwork/minf/pvt-ucx1022/test/apps/iodemo/io_demo.cc:2867
11 0x0000000000022555 __libc_start_main()  ???:0
12 0x0000000000404d9c _start()  ???:0
=================================

Reply:
maybe something like:
```
if (ucp_wireup_ep_has_next_ep(cm_wireup_ep) {  // add new function ucp_wireup_ep_has_next_ep to wireup_ep.c
    /* If we failed to create CM ep, next_ep is not initialized */
    ucp_wireup_ep_destroy_next_ep(cm_wireup_ep);
}
```

Reply:
please help review the change